### ISSUE TITLE
Covers are not displayed on the carousel

### ISSUE THREAD

**Issue by dcapillae** [internetarchive/openlibrary#3595]
Some covers are not displayed on the carousel. 

### Evidence / Screenshot
A custom carousel
![screenshot](https://user-images.githubusercontent.com/7125952/87861049-c9452d80-c942-11ea-8961-eb8fd56b927d.png)

Publisher's page
![screenshot](https://user-images.githubusercontent.com/7125952/87861529-686c2400-c947-11ea-9145-c11535460a2f.png)

### Relevant url?
A page with several carousels where the covers are not displayed:
https://openlibrary.org/collections/cascaborra-historia-de-espana-en-vinetas

The bug also occurs in the [publisher's page](https://openlibrary.org/publishers/Cascaborra), although not in the same way (?)

### Details

- **Logged in (Y/N)?** Y
- **Browser type/version?** Firefox 68.10
- **Operating system?** Debian GNU/Linux
- **Environment (prod/dev/local)?** prod 

### Proposal & Constraints
Perhaps it is related to the fact that the books no longer have a own cover/page for the work, only for editions. Could it be due to [line 22](https://github.com/internetarchive/openlibrary/blob/83e522de812c8d79e9f49d7e2cdc2fe3f5b11eb5/openlibrary/templates/books/custom_carousel.html#L22)? Perhaps by making a small change (`/book/` to `/olid/`):

`$ cover_url = '%s/b/olid/%s-M.jpg'%(cover_host, book.get('cover_edition_key'))`

**Comment by cdrini** [comment]
That proposal looks spot on; nice work @dcapillae ! If you can open a PR (even through GH), I can put it on dev.openlibrary.org for testing.

**Comment by guyjeangilles** [comment]
Strangely, prod looks fixed, but dev is broken as the issue describes for the publisher. The custom carousel is broken for me on both prod and dev.
![dev_ol_3595](https://user-images.githubusercontent.com/7256157/88011126-0f96ba00-cb06-11ea-86cc-1f6a65803780.png)
![3595_ol_prod](https://user-images.githubusercontent.com/7256157/88011252-656b6200-cb06-11ea-8816-0a143c3fcee2.png)

**Comment by guyjeangilles** [comment]
This is still a bug on prod and dev
![ol_3595_2](https://user-images.githubusercontent.com/7256157/88124479-c7869e80-cbbc-11ea-889f-02fcd91f9bb1.png)

**Comment by tabshaikh** [comment]
@guyjeangilles The books which do not have a cover is due to the work not having a cover, should this issue be closed then?

**Comment by dcapillae** [comment]
@tabshaikh @guyjeangilles  I don't think so. I don't think it should be closed. The carousels still don't work. If a work doesn't have a cover, this point should be fixed. With the new UI, it's not possible to add covers to works, only to editions.

**Comment by Sabreen-Parveen** [comment]
Is someone working on this issue? If not, I want to work on this issue.

**Comment by cdrini** [comment]
This should now be fixed by edition-aware carousels. Any records missing a cover can be fixed by adding covers to the corresponding edition :+1:
